File: src/api/EmployeeService.js
import axios from "axios";

const BASE_URL = "http://localhost:8080/api";

class EmployeeService {

  getEmployees(page = 0, size = 5) {
    return axios.get(`${BASE_URL}/employees?page=${page}&size=${size}`);
  }

  searchEmployees(keyword, page = 0, size = 5) {
    return axios.get(`${BASE_URL}/employees/search?keyword=${keyword}&page=${page}&size=${size}`);
  }

  filterByDepartment(department, page = 0, size = 5) {
    return axios.get(`${BASE_URL}/employees/filter?department=${department}&page=${page}&size=${size}`);
  }

  addEmployee(employee) {
    return axios.post(`${BASE_URL}/employees`, employee);
  }

  updateEmployee(id, employee) {
    return axios.put(`${BASE_URL}/employees/${id}`, employee);
  }

  deleteEmployee(id) {
    return axios.delete(`${BASE_URL}/employees/${id}`);
  }

  
  getDashboard() {
    return axios.get(`${BASE_URL}/dashboard`);
  }
}

export default new EmployeeService();

File: employeeapp/src/main/java/in/MeghaP/service/EmployeeService.java
package in.MeghaP.service;

import in.MeghaP.model.Employee;
import in.MeghaP.repository.EmployeeRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.stereotype.Service;

@Service
public class EmployeeService {

    @Autowired
    private EmployeeRepository employeeRepository;

    public Employee addEmployee(Employee e) {
        return employeeRepository.save(e);
    }

    public Page<Employee> getEmployees(int page, int size) {
        return employeeRepository.findAll(PageRequest.of(page, size));
    }

    public Page<Employee> searchEmployees(String keyword, int page, int size) {
        return employeeRepository.search(keyword, PageRequest.of(page, size));
    }
    public Page<Employee> filterByDepartment(String department, int page, int size) {
        return employeeRepository.findByDepartmentIgnoreCase(
                department,
                PageRequest.of(page, size)
        );
    }


    public Employee updateEmployee(Long id, Employee newData) {
        return employeeRepository.findById(id).map(emp -> {
            emp.setName(newData.getName());
            emp.setEmail(newData.getEmail());
            emp.setPhone(newData.getPhone());
            emp.setDepartment(newData.getDepartment());
            emp.setSalary(newData.getSalary());
            return employeeRepository.save(emp);
        }).orElse(null);
    }

    public void deleteEmployee(Long id) {
        employeeRepository.deleteById(id);
    }
}
